src/pip/_internal/utils/misc.py

```diff
@@ -745,6 +745,7 @@ def read_text_file(filename):
         break
 
     assert type(data) != bytes  # Latin1 should have worked.
+    fp.close()
```

Comment:
This shouldn't be needed as ```fp``` is automatically closed by the ```with``` statement. Have you any evidence that there's an actual issue here?